### ISSUE TITLE
Extending rules template to support ORIGINAL DEST

### ISSUE THREAD

**Issue by apiening** [mrlesmithjr/ansible-shorewall#19]
I have the requirement to set the field **ORIGINAL DEST** in **/etc/shorewall/rules** in order to be able to match specific target IP's on a server.
This field is the seventh parameter in a rule definition, right after **SOURCE PORT(S)** which is also not filled by the current template.

As far as I see, there is just the line nr. **6** in **templates/etc/shorewall/rules.j2** to be modified in order to achieve this:
```
{{ rule.action }}   {{ rule.source }}   {{ rule.dest }}   {{ rule.proto }}   {{ rule.dest_ports|join (',') }}   {{ rule.source_ports|join (',') }}   {{ rule.original_dest }}
```

Another idea (that has also been addressed in the pull request https://github.com/mrlesmithjr/ansible-shorewall/pull/18) is to add support for comments.

This could be done by adding these three lines right before the rule line inside the loop:
```
{% if rule.comment|length %}
#{{ rule.comment }}
{% endif %}
``` 

Of course this is not a issue, it is more a suggestion. I think it would be easy to apply and should not break something because if the variables are empty the template expands to the same rule line.
What do you think?

Btw. just want to say thank you for your ansible role. It is the most often used role in my playbooks.

**Comment by rogerniesten** [comment]
I’ve added the comment by changing the affected in in the roles.j2 template:

{{ rule.action }}   {{ rule.source }}   {{ rule.dest }}   {{ rule.proto }}   {{ rule.dest_ports|join (',') }}    {% if rule.comments is defined %}# {{ rule.comments }} {% endif %}

When comments field is added tot the dictionary, it will be added to the rules file.

**Comment by apiening** [comment]
> I’ve added the comment by changing the affected in in the roles.j2 template: {{ rule.action }} {{ rule.source }} {{ rule.dest }} {{ rule.proto }} {{ rule.dest_ports|join (',') }} {% if rule.comments is defined %}# {{ rule.comments }} {% endif %} When comments field is added tot the dictionary, it will be added to the rules file.

Hi Roger,
which `rules.j2` template did you update? Can't see the change in the master branch and this seems to be the only branch.

I would prefer the optional comment on top of the rule line instead at the end, especially when a lot of options are set (like the suggested `ORIGINAL DEST` field) the rules get pretty long which makes them harder to read with a comment at the end that would potentially be wrapped to a new line by the editor. But that's just cosmetics.

What about merging these changes into master?

**Comment by rogerniesten** [comment]
Hi Andreas,

I had to make some more changes specifically for our environment and didn’t cherry pick any parts as potential pull requests. I’ve committed them in our on fork: https://github.com/MaastrichtUniversity/ansible-shorewall/commit/a46898a8594f5c7213ea6733bba9892a1c172b30
Feel free to add these changes to your own fork or mrlesmithjr if you like.

**Comment by apiening** [comment]
Hi Roger,

ok I see.

If the extension to support `ORIGINAL DEST` is not merged, I would need to create a fork because I really need this setting to manage more complex rules on a few systems.
The best option from my perspective is to have one central repository for this role which is updated and improved over time, compared to end up with several forks with feature fragmentation.

@mrlesmithjr What do you think? Is it possible to update the `rules.j2` template file, or would you prefer a pull request?

With kind regards

Andreas

**Comment by mrlesmithjr** [comment]
@apiening @rogerniesten Glad both of you are getting some use out of this role. And I'd absolutely be open to any pull requests to add additional functionality, etc.

**Comment by apiening** [comment]
@mrlesmithjr Thank you. I've just committed a pull request (https://github.com/mrlesmithjr/ansible-shorewall/pull/20).

I have commented the change in the PR and only one file is affected (**rules.j2**).
I tried to make sure that the changes does not affect the content of the rule file when the new fields are unset by using defaults for the rule line. So I can't see a situation where this change might break the config with existing config-variables.
I did some tests with **shorewall 5.1.12.2 on Ubuntu 18.04.4 LTS** and **shorewall 5.2.3.4 on Ubuntu 20.04 LTS** without any issues.

I would be happy if you would accept the PR or discuss if anything is unclear or need to be changed.
